packages/table/src/table.scss

```diff
@@ -88,6 +88,9 @@ $menu-z-index: $row-z-index + 1 !default;
 }
 
 .bp-table-row-headers {
+  // let header's height resize based on height of child element
+  display: table;
```

Comment:
No; at least I don't think so -- display: flex is parent behavior, flex behavior defines behavior of the child. And display: table does behavior of children again. 

That said: if you can figure out how to make the row headers take on the size of its children the way the column headers does, using flexbox rather than this, I would prefer to use that instead.